### ISSUE TITLE
Improve animations

### ISSUE THREAD

**Issue by jancborchardt** [stefan-niedermann/nextcloud-notes#51]
Currently when opening and closing a note, there’s this strange up-and-down animation. This is quite jarring and doesn’t make a lot of sense spacially.

What people are used to more is a left-right animation. The more detailed view slides in from the right. Pressing the back button slides back to the left to the list.

**Comment by HeaDBanGer84** [comment]
I startet to implement Animationtions in my fork like discribed http://android-developers.blogspot.de/2014/10/implementing-material-design-in-your.html in section Authentic Motion

see commit:
https://github.com/HeaDBanGer84/OwnCloud-Notes/commit/ea43f085da8f5feae56c029fb6053aa5750d5ba1 and
https://github.com/HeaDBanGer84/OwnCloud-Notes/commit/26a836df0bd670ce8739afa57fb6cca071455edd

**Comment by stefan-niedermann** [comment]
Hi @HeaDBanGer84,

thank you for starting working on this!

any chance to move you back to the ListLayout instead of the CardView Layout? I think it will be very difficult in the future to merge back all these changes when the two projects are using different layouts everywhere...

**Comment by jancborchardt** [comment]
Yeah, please use the list view layout so we are on a common base. :) Also, if you can, a gif or video of the animation would be cool! :)

**Comment by HeaDBanGer84** [comment]
If you compile current version, you see that i don't use the cardlayout. It can be activated, with some flags which just need 3 extra lines of code.
simple demo:
![transitions](https://cloud.githubusercontent.com/assets/11348045/12450042/99e2da30-bf81-11e5-8c5c-2f5aadda416b.gif)

**Comment by jancborchardt** [comment]
While the animations are playful this way, they need to look absolutely perfect. And currently it seems a bit jerky. Can we do better there?

**Comment by jancborchardt** [comment]
Also, the look of the list view seems to have a regression. It looks kinda old-style with very visible horizontal lines etc. It should look like a nice Material design list instead.

**Comment by HeaDBanGer84** [comment]
I record this from an emulator and converted it to an animated .gif. It looks a lot better on real Hardware. The seperators  are down in a dedicated .xml file. But I was not able to find the perfekt color and thickness for it. if @stefan-niedermann knows better just adjust the values in /res/drawable/line_divider.xml

**Comment by stefan-niedermann** [comment]
Sorry, but your fork is definitely too much away from where i want to go with the app.
1.) The design regressions @jancborchardt mentioned are no problems, i would fix that, but
2.) The animation... it fades the whole screen instead of just moving one activity over another. this is not the wanted behaviour
3.) The Sections are not working anymore, too. i think this problem is caused by the switch from listlayout to cards.

Of course you can feel free to develop your own fork, the app is open source. But i will not merge these changes back. Sorry.

**Comment by jancborchardt** [comment]
I can totally understand @stefan-niedermann here and support his decision.

Since I have seen too many small open source apps fail because there were too few people working on it together, I would pledge for collaboration here instead of a fork, please. :) @stefan-niedermann @HeaDBanGer84

**Comment by HeaDBanGer84** [comment]
If it is possible to contribute it would continue. But I have to keep the app usable for me
@stefan-niedermann 
 1. completely agree, I wasn't able to find the right color and thickness for this seperator, becaus RecyclerView has a very defrent approach to it than the classic ListView
 2. Take the suggtion from developer blog and if you watch that https://www.youtube.com/watch?v=RhiPJByIMrM is exatly what google recommended, to guide the user.
 3. The section Problem appears from the ViewHolder-concept (which actually gives the performance-boost) of the RecyclerView which makes it much more difficult to layout to completely different views. I can try to think about something using inheritance to make it possible, but not sure if this is working. the only idea I had to do this is to do multiple RecyclerViews which creates a big overhead.
 4. I used a completly diffrent layout for the list so, there are really no cardviews used at all. 

Hope we will find a solution, I still not gave up to find a compromise.

**Comment by stefan-niedermann** [comment]
Thanks.
> The section Problem appears from the ViewHolder-concept

Please have a look at my [RecyclerView-Branch](https://github.com/stefan-niedermann/OwnCloud-Notes/commits/RecyclerView) where i was able to use ViewHolder-Concept *and* Sections. This branch is nearly ready to merge in master, there are only a view Bugs left.
- [ ] Creating a new Note creates the note two times
- [ ] Deleting a Note via long-click selects another Note after Deletion

I will keep you updated when this branch is ready and merged into master. Then i think we will have a clear status and can continue on working on the *real* issues.

In the meantime i will have a look at the Youtube-video.

@jancborchardt sorry for flooding this issue, i will maybe open a new one, but this discussion about collaboration is needed ;)

**Comment by jancborchardt** [comment]
@stefan-niedermann I’m totally fine with you two discussing this here. :) It’s a very important topic.

**Comment by jakejoh** [comment]
I think this issue can be closed?
The animation for opening/closing notes is there and the fork seems to be private and/or discontinued.

**Comment by stefan-niedermann** [comment]
Not sure about this one. @HeaDBanGer84 was right, you know? Moving parts of the UI as a transition is correct Material Design, but the font-style of the "source" and the "target" is very different, so it doesn't feel very natural.

I think this approach applies better to labels, images or cards.

In the end i don't think that i don't want to go this way, but i feel we can improve the animation though.
I played around a bit with "explode", but it shouldn look *too* fancy in the end 🤔

**Comment by jancborchardt** [comment]
The current animation is much much better, so this issue is not really current anymore.

The only thing is that I’d still say is we should go for a left/right animation on opening and closing a note, cause it reflects the hierarchy. But @stefan-niedermann if you prefer the current style I’m fine with it and we can close this. :)

**Comment by stefan-niedermann** [comment]
I'll keep it open, because i want to test the left/right animation on opening / closing notes.

**Comment by stefan-niedermann** [comment]
**TL;DR: We will keep the current state.**

Sooooo coming back to this old issue one more time. I have tried very many different approaches to animate opening and closing notes from the list with and without shared elements.

### Shared elements (suggested by @HeaDBanGer84)

Doesn't work well because basically everything is text. This works when images, labels or other boxes, maybe even buttons are floating around (and we actually use that in the Deck app), but for moving text around it looks ugly.

Other drawbacks:

- In the list we can have multiple lines "compressed" to one line as excerpt, but we cannot animate this change to split up again to multiple lines in the editor.
- The category would need a huge transition from the box (which also looks different in dark mode) to a subtitle of the toolbar
- If we use multiple shared elements, they will cross the lines of each other
![never cross the streams](https://memecrunch.com/meme/BXKV9/don-t-cross-the-streams/image.jpg)

### Explode

Doesn't work well because we have in the target activity (the editor) only two views: The toolbar at the top and the editor at the bottom. This means basically that there is one view floating from top to bottom and one from bottom to top. It looks like a glitch.

### Fade

This one was actually one of the transitions that *might* work, pretty smooth, but since the two activity look quiet the same (header, much text content in the rest oft he available space) i felt like we should take a bit more attention so the user actually *notices* that something changes. It was simply too "stealthy" - i wondered "huh? Why doesn't the editor loa-.. aah, well, i am already looking at the editor".

### Left / Right (proposed by @jancborchardt)

Also tried this one. Even if i were able to solve the delay of the toolbar (wtf google?), it doesn't feel good. It crashes a bit with the `swipe 2 delete` / `swipe 2 favorite` features and i catched even myself to want to swipe a note to open it (yes i thought about actually implementing this - but this is no good pattern and i don't know it from anywhere else)

PS.: Ignore the fact that this app looks quiet different from what is currently released, i am just experimenting a little bit :innocent: 

![left-right-animation](https://user-images.githubusercontent.com/4741199/83880869-2644a580-a740-11ea-9e2b-13afe05a232d.gif)

**Comment by jancborchardt** [comment]
Just for an additional point about the left/right animations. I agree that a pure left/right animation doesn’t look so good and confuses regarding what can be done with swiping. But if you check how e.g. Signal does it, they have a really elegant way of switching between the 2 views, where the conversation list is not fully pushed to the left, but more shoved into the background (below the other view in the Z axis) and darkened.

**Comment by stefan-niedermann** [comment]
Can you post a GIF or a video please which shows opening the "details" and going back to the list?

I searched youtube, but couldn't find a sample...
